server/task_executor_schema_update_ghost_sync.go

```diff
@@ -241,23 +246,18 @@ func (exec *SchemaUpdateGhostSyncTaskExecutor) runGhostMigration(_ context.Conte
 	}(ctx)
 
 	go func() {
-		if err := migrator.Migrate(); err != nil {
+		err := migrator.Migrate()
+		if err != nil {
 			log.Error("failed to run gh-ost migration", zap.Error(err))
-			// `migrator.Migrate` can return an error.
-			// 1. If the error is returned before closing `syncDone`,`runGhostMigration` will receive from `syncError`.
-			// 2. If it is returned after `syncDone` is closed, `runGhostMigration` would have received from `syncDone` and returned,
-			// and there's no consumer for `syncError`, so we must send to `syncError` without blocking.
-			select {
-			case syncError <- fmt.Errorf("failed to run gh-ost migration, error: %w", err):
-			default:
-			}
 		}
+		migrationError <- err
 	}()
 
 	select {
 	case <-syncDone:
+		server.TaskScheduler.sharedTaskState.Store(task.ID, ghostState{migrator: migrator, errCh: migrationError})
 		return true, &api.TaskRunResultPayload{Detail: "sync done"}, nil
-	case err := <-syncError:
+	case err := <-migrationError:
```

Comment:
Please comment on the possible happening order of `syncDone` and `migrationError`.
For example:
1. `migrationError` happens before `syncDone`, the sync task fails before the gh-ost table is in sync with the original table.
2. ...